src/query/service/src/interpreters/interpreter_show_columns.rs

```diff
@@ -0,0 +1,258 @@
+}
+
+impl ShowColumnsInterpreter {
+    pub fn try_create(ctx: Arc<QueryContext>, plan: ShowColumnsPlan) -> Result<Self> {
+        Ok(ShowColumnsInterpreter { ctx, plan })
+    }
+}
+
+#[async_trait::async_trait]
+impl Interpreter for ShowColumnsInterpreter {
+    fn name(&self) -> &str {
+        "ShowCreateTableInterpreter"
+    }
+
+    fn schema(&self) -> DataSchemaRef {
+        self.plan.schema()
+    }
+
+    #[async_backtrace::framed]
+    async fn execute2(&self) -> Result<PipelineBuildResult> {
+        let tenant = self.ctx.get_tenant();
+        let catalog = self.ctx.get_catalog(self.plan.catalog.as_str())?;
+
+        let table = catalog
+            .get_table(tenant.as_str(), &self.plan.database, &self.plan.table)
+            .await?;
+
+        let engine = table.engine();
+        let fields = if engine == VIEW_ENGINE {
+            if let Some(query) = table.options().get(QUERY) {
+                let mut planner = Planner::new(self.ctx.clone());
+                let (plan, _) = planner.plan_sql(query).await?;
+                let schema = infer_table_schema(&plan.schema())?;
+                schema.fields().clone()
+            } else {
+                return Err(ErrorCode::Internal(
+                    "Logical error, View Table must have a SelectQuery inside.",
+                ));
+            }
+        } else {
+            table.schema().fields().clone()
+        };
+
+        let fields_len = fields.len();
+        let mut names: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut data_types: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut is_nullables: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut exprs: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut extras: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut keys: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        // full
+        let mut collations: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut privileges: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+        let mut comments: Vec<Vec<u8>> = Vec::with_capacity(fields_len);
+
+        // Append columns.
+        {
+            for field in fields.into_iter() {
+                names.push(field.name().clone().into_bytes());
+                let data_type = field.data_type().remove_recursive_nullable().sql_name();
+                data_types.push(data_type.into_bytes());
+
+                let mut default_expr = "".to_string();
+                if let Some(expr) = field.default_expr() {
+                    default_expr = expr.to_string();
+                }
+                exprs.push(default_expr.into_bytes());
+                if field.is_nullable() {
+                    is_nullables.push("YES".to_string().into_bytes());
+                } else {
+                    is_nullables.push("NO".to_string().into_bytes());
+                }
+                extras.push("".to_string().into_bytes());
```

Comment:
this column seems always empty, how about use `BlockEntry` with a `Scalar` value. like:
```rust
BlockEntry {
    data_type: DataType::String,
    value: Value::Scalar(Scalar::String(vec![])),
}
```

Reply:
If do this, can not ensure that the sequence is the same as the schema?